### PR TITLE
Expand substitution documentation

### LINKED ISSUE #4078: Variable substitution in href of <a> tags?
I'd love to see the ability to do at least simple variable substitution in the `href` of an `<a>` tag.  specifically if I could get `CLIENT_ID` transmitted it would let me track sessions between the a cdn and regular site.  One of the big gripes we're getting from publishers is they have no idea if AMP is generating more or longer sessions or if users are just swiping in the carousel to another pubs content.  

I realize that some of the variables are real time so that would involve intercepting clicks and that bring it's own issues.  However  if we could just get the static stuff that would still be a huge win.

### PR DESCRIPTION
**Pull request by rudygalfi**
More specifics on different features supporting substitutions and various restrictions.
Full documentation on using link substitutions.

Closes #4078.

### CODE REVIEW

File: spec/amp-var-substitutions.md [hunk 1]
```diff
@@ -18,16 +18,101 @@ limitations under the License.
...
+  </tr>
+  <tr>
+    <td width="25%">Form inputs<br><a href="https://github.com/ampproject/amphtml/blob/master/extensions/amp-form/amp-form.md#variable-substitutions">Detailed documentation</a></td>
+    <td width="25%">Requests must be HTTPS URLs (not a requirement specific to variable substitutions)</td>
+    <td width="25%">Yes, via space-delimited attribute <code>data-amp-replace</code> (see below)</td>
+    <td width="25%">See the section on “substitution timing” below for further notes.</td>
+  </tr>
+</table>
+
+**Substitution timing**: Variable substitutions that are dependent on a user action like links and form inputs may not occur if the value has not yet been computed.
```
Comment (bpaduch) [inline spec/amp-var-substitutions.md]: Make this a heading 3 (`###`)

File: spec/amp-var-substitutions.md [hunk 2]
```diff
@@ -18,16 +18,101 @@ limitations under the License.
...
+    <td width="25%">
+      HTTPS URL and URL matching either:
+      <ul>
+        <li>Page’s source origin</li>
+        <li>Page’s canonical origin</li>
+        <li>An origin whitelisted via the <code>amp-link-variable-allowed-origin</code> <code>meta</code> tag</li>
+      </ul>
+    </td>
+    <td width="25%">Yes, via space-delimited attribute <code>data-amp-replace</code> (see below)</td>
+    <td width="25%">Only these variables are supported: <code>CLIENT_ID</code> and <code>QUERY_PARAM</code>.<br>See the section on “substitution timing” below for further notes.</td>
```
Comment (bpaduch) [inline spec/amp-var-substitutions.md]: Link to the "Section Timing" heading

File: spec/amp-var-substitutions.md [hunk 3]
```diff
@@ -18,16 +18,101 @@ limitations under the License.
...
+  </tr>
+  <tr>
+    <td width="25%">Form inputs<br><a href="https://github.com/ampproject/amphtml/blob/master/extensions/amp-form/amp-form.md#variable-substitutions">Detailed documentation</a></td>
+    <td width="25%">Requests must be HTTPS URLs (not a requirement specific to variable substitutions)</td>
+    <td width="25%">Yes, via space-delimited attribute <code>data-amp-replace</code> (see below)</td>
+    <td width="25%">See the section on “substitution timing” below for further notes.</td>
```
Comment (bpaduch) [inline spec/amp-var-substitutions.md]: Link to the "Section Timing" heading instead of using below.

File: spec/amp-var-substitutions.md [hunk 4]
```diff
@@ -18,16 +18,101 @@ limitations under the License.
...
+* `CLIENT_ID` is available once it has been computed. This can be accomplished through use by another feature such as `amp-analytics` or `amp-pixel`. Note that `CLIENT_ID` may also be blocked on an `amp-user-notification` that is pending acceptance.
+* Asynchronously resolved variables are not available
+
+## Variable substitution in links
+
+Variable substitution is available in links, i.e. `<a href="..."></a>`.
+
+Only these variables are supported:
+* `CLIENT_ID`
+* `QUERY_PARAM(argument)`.
```
Comment (bpaduch) [inline spec/amp-var-substitutions.md]: Remove period

File: spec/amp-var-substitutions.md [hunk 5]
```diff
@@ -18,16 +18,101 @@ limitations under the License.
...
+        <li>An origin whitelisted via the <code>amp-link-variable-allowed-origin</code> <code>meta</code> tag</li>
+      </ul>
+    </td>
+    <td width="25%">Yes, via space-delimited attribute <code>data-amp-replace</code> (see below)</td>
+    <td width="25%">Only these variables are supported: <code>CLIENT_ID</code> and <code>QUERY_PARAM</code>.<br>See the section on “substitution timing” below for further notes.</td>
+  </tr>
+  <tr>
+    <td width="25%">Form inputs<br><a href="https://github.com/ampproject/amphtml/blob/master/extensions/amp-form/amp-form.md#variable-substitutions">Detailed documentation</a></td>
+    <td width="25%">Requests must be HTTPS URLs (not a requirement specific to variable substitutions)</td>
+    <td width="25%">Yes, via space-delimited attribute <code>data-amp-replace</code> (see below)</td>
```
Comment (bpaduch) [inline spec/amp-var-substitutions.md]: Link to "Per-use opt-in" section instead of using below

File: spec/amp-var-substitutions.md (hunk 2 above)
Comment (rudygalfi) [inline spec/amp-var-substitutions.md]: done

File: spec/amp-var-substitutions.md (hunk 5 above)
Comment (rudygalfi) [inline spec/amp-var-substitutions.md]: done

File: spec/amp-var-substitutions.md (hunk 3 above)
Comment (rudygalfi) [inline spec/amp-var-substitutions.md]: done

File: spec/amp-var-substitutions.md (hunk 1 above)
Comment (rudygalfi) [inline spec/amp-var-substitutions.md]: done

File: spec/amp-var-substitutions.md (hunk 4 above)
Comment (rudygalfi) [inline spec/amp-var-substitutions.md]: done

**Comment by rudygalfi** [comment]
Thanks for the review! Should be all set now.

**Pull request merged**